src/EFCore.PG/Query/Internal/NpgsqlSetOperationTypeResolutionCompensatingExpressionVisitor.cs

```diff
@@ -0,0 +1,152 @@
+using Microsoft.EntityFrameworkCore.Query.Internal;
+
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.Query.Internal;
+
+public class NpgsqlSetOperationTypeResolutionCompensatingExpressionVisitor : ExpressionVisitor
```

Comment:
@smitpatel this is the visitor which fixes the PG set operation thing (#2366). As it's a bit tricky it would be great if you could take a look (but only if you have time).